Soycode pgpintegration

soycode opened the pull request:
Passes (after https://github.com/freedomjs/freedom-for-node/pull/27)

willscott commented on package.json:
```diff
@@ -12,13 +12,15 @@
     "es6-promise": "^1.0.0",
-    "freedom": "^0.6.11",
+    "freedom": "^0.6.18",
+    "freedom-for-node": "^0.2.10",
```
is this the as-of-yet-unpublished version?

soycode replied:
Both freedom and freedom-for-node look to be the latest on npmjs.com.

willscott commented on spec/integration/pgpapi.spec.js:
```diff
@@ -0,0 +1,31 @@
+/*globals describe, beforeEach, require, expect, it */
+/*jslint indent:2,white:true,sloppy:true,node:true*/
```
would prefer not having the ```white:true,sloppy:true``` and holding our selves to stricter style validation.

soycode replied:
Yeah had that in from copy-pasting from other files, removed from all the files and they still lint fine. Will add that plus a few other things in next commit.

willscott commented:
after getting the node version you depend on, :+1: to merge.

soycode commented:
Got travis stuff working, passes in node 0.10 but not in 0.11.  https://travis-ci.org/freedomjs/freedom-pgp-e2e/jobs/47710038

From that output it's not clear to me what is failing, it's just timing out and complaining about console.log. Let me know if it makes sense to you, else I may remove 0.11 for now.

willscott commented:
Add some console.log's in a commit, and travis should run again, and that
will let you see if messages are actually getting sent across in 0.11.
On Tue Jan 20 2015 at 2:26:04 PM soycode <notifications@github.com> wrote:

> Got travis stuff working, passes in node 0.10 but not in 0.11.
> https://travis-ci.org/freedomjs/freedom-pgp-e2e/jobs/47710038
>
> From that output it's not clear to me what is failing, it's just timing
> out and complaining about console.log. Let me know if it makes sense to
> you, else I may remove 0.11 for now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freedomjs/freedom-pgp-e2e/pull/20#issuecomment-70747204>
> .
>

soycode commented:
So I added logging and pushed it until I narrowed it to a decryption timeout: https://travis-ci.org/freedomjs/freedom-pgp-e2e/jobs/47718765#L264

Added more logging deeper in to the actual module code and ended up with a fairly inscrutable error: https://travis-ci.org/freedomjs/freedom-pgp-e2e/jobs/47723365#L261

I tried removing logging of anything besides simple static strings but it still gives that error, so I still can't get at whatever the underlying issue is. Anyway, I'm making this comment for my own record as well, will revisit w/fresh mind in a bit, opinions of course are welcome.

Aside - grunt-jasmine-node is stale, I switched to grunt-jasmine-node2 fixed the util.print issue so we can see more details.

willscott commented:
cool on the grunt-jasmine-node2 thing, would be great to get a pull request
in freeom-for-node on that migration as well :)

On Tue Jan 20 2015 at 4:22:32 PM soycode <notifications@github.com> wrote:

> So I added logging and pushed it until I narrowed it to a decryption
> timeout:
> https://travis-ci.org/freedomjs/freedom-pgp-e2e/jobs/47718765#L264
>
> Added more logging deeper in to the actual module code and ended up with a
> fairly inscrutable error:
> https://travis-ci.org/freedomjs/freedom-pgp-e2e/jobs/47723365#L261
>
> I tried removing logging of anything besides simple static strings but it
> still gives that error, so I still can't get at whatever the underlying
> issue is. Anyway, I'm making this comment for my own record as well, will
> revisit w/fresh mind in a bit, opinions of course are welcome.
>
> Aside - grunt-jasmine-node is stale, I switched to grunt-jasmine-node2
> fixed the util.print issue so we can see more details.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freedomjs/freedom-pgp-e2e/pull/20#issuecomment-70762308>
> .
>

soycode commented:
https://github.com/freedomjs/freedom-social-xmpp/pull/83

Unfortunately the freedom-for-node tests fail when switching to grunt-jasmine-node2 - I'll investigate a bit further, but we may also just want to switch to grunt-karma/karma-jasmine. https://github.com/jasmine-contrib/grunt-jasmine-node/issues/60

ryscheng commented:
+1. I haven't figured out how to get Karma to run in a node environment
yet, but seems worth it, considering how many issues we have with
jasmine-node

On Wed, Jan 21, 2015 at 10:30 AM, soycode <notifications@github.com> wrote:

> freedomjs/freedom-social-xmpp#83
> <https://github.com/freedomjs/freedom-social-xmpp/pull/83>
>
> Unfortunately the freedom-for-node tests fail when switching to
> grunt-jasmine-node2 - I'll investigate a bit further, but we may also just
> want to switch to grunt-karma/karma-jasmine.
> jasmine-contrib/grunt-jasmine-node#60
> <https://github.com/jasmine-contrib/grunt-jasmine-node/issues/60>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freedomjs/freedom-pgp-e2e/pull/20#issuecomment-70893387>
> .
>
